### ISSUE TITLE
Zoom, and Cropper size.

### ISSUE THREAD

**Issue by alexdpts** [Norserium/vue-advanced-cropper#147]
How can i bind Zoom on mouseWheel+key (eq. Shift+mouseWheel)?
How can i Resize Cropper height after zooming image to image size after zooming (eq. if zoomed image height > Cropper height)?

**Comment by Norserium** [comment]
Hello, @alexdpts!

> How can i bind Zoom on mouseWheel+key (eq. Shift+mouseWheel)?

And prevent zoom on mouse wheel?

> How can i Resize Cropper height after zooming image to image size after zooming (eq. if zoomed image height > Cropper height)?

Could you provide any example? It's not clear what you want to get.

**Comment by luckydonald** [comment]
Yes, that automatic zooming is a unwanted behavior here as well.

**Comment by Norserium** [comment]
> Yes, that automatic zooming is a unwanted behavior here as well.

@luckydonald , what an automatic zooming?

**Comment by alexdpts** [comment]
> And prevent zoom on mouse wheel?
Bind zoom on key+mouseWheel, and leave normal page scroll on mouseWheel.

> Could you provide any example? It's not clear what you want to get.
For example:https://norserium.github.io/vue-advanced-cropper/introduction/getting-started.html#minimal-working-example
Cropper container has fixed size. When i zoom image, container size not change. As result, if zoomed image size over than cropper container size, i can't select parts of image that are outside of container.
This is why i need:
1. Zoom on key+mousewheel
2. Normal page scroll on mousewheel
3. Resize (fit) cropper container to zoomed image
for select image part to crop, when image eg zoomed to maximum.

Sorry, for my tеrrible english

**Comment by Norserium** [comment]
@alexdpts, alas, it's not supported native (just like in the other available cropper alternatives). You can try to get around this with some sort of a workaround, but I need more details to advice something.

What if the container can't become larger? For example, it has the same width as the window width?

**Comment by Norserium** [comment]
@alexdpts, alas, there are problems with implementation the last requirement. It looks like it impossible to do right now without the significant code changes.

For the other ones I wrote [the simple workaround](https://codesandbox.io/s/vue-advanced-cropper-resize-shift-key-8s0mo?file=/src/App.vue).

**Comment by alexdpts** [comment]
> alas, it's not supported native (just like in the other available cropper alternatives). You can try to get around this with some sort of a workaround, but I need more details to advice something.
> 
> What if the container can't become larger? For example, it has the same width as the window width?

I try to show on some simple pictures:
1. at start we have page with cropper container contains image, yellow rectangle - area we need  
![01](https://user-images.githubusercontent.com/17212759/126087829-6b47930b-5817-4ad1-a39d-3152f8526071.jpg)
2. we roughly choose area with stensil 
![02](https://user-images.githubusercontent.com/17212759/126087954-065b2cb1-5a4b-4a35-b466-ce8800d26e17.jpg)
3. For some precission selection zoom image. As a zoom result, we get, that yellow area is out of cropper container 
![03](https://user-images.githubusercontent.com/17212759/126088128-6e2e204f-763f-495f-8e97-de4f3c460bdb.jpg)
Yes, we can move the image so that the yellow zone is inside the container, but what if we need to crop along the green borders with some precision 
![04](https://user-images.githubusercontent.com/17212759/126088285-7e6feb7f-210a-420b-adca-1914ab1e2fdc.jpg)

**Comment by Norserium** [comment]
@luckydonald thanks for the detailed response. I understand the concept, but it looks unusual. Did you encounter any real cropper example that implements it?

**Comment by alexdpts** [comment]
> thanks for the detailed response. I understand the concept, but it looks unusual. Did you encounter any real cropper example that implements it?

i try https://github.com/timtnleeProject/vuejs-clipper that implements this (with some css corrections), but yjat can't flip image.

**Comment by Norserium** [comment]
@alexdpts, I'm sorry, but I didn't see the real difference in the behavior of my cropper and `vuejs-clipper`. Could you give the link to the example, that can't be implemented in my cropper?

**Comment by alexdpts** [comment]
> @alexdpts, I'm sorry, but I didn't see the real difference in the behavior of my cropper and `vuejs-clipper`. Could you give the link to the example, that can't be implemented in my cropper?

I make example. See at http://f0508697.xsph.ru/ 

try load any big vertical image.

at left hand - vue advanced cropper
at right - vuejs-clipper

**Comment by Norserium** [comment]
@alexdpts, thank you very much for the example. Your issue is now clear to me.

The cropper just tries to be responsive and itself in the limited area (`vacCell`). It is the useful feature for many different use-cases.

You can redefine this behavior by [`updateBoundaries` prop](https://norserium.github.io/vue-advanced-cropper/components/cropper.html#defaultboundaries) to achieve the exactly same behavior. There is [the example](https://codesandbox.io/s/vue-advanced-cropper-custom-boundaries-algorithm-d8knq).

**Comment by alexdpts** [comment]
> @alexdpts, thank you very much for the example. Your issue is now clear to me.
> 
> This cropper just tries to be responsive and fit itself in the limited area (`vacCell`). It is the useful feature for many different use-cases.
> 
> You can redefine this behavior by setting [`updateBoundaries`](https://norserium.github.io/vue-advanced-cropper/components/cropper.html#defaultboundaries) prop to achieve the exactly same behavior. There is [the example](https://codesandbox.io/s/vue-advanced-cropper-custom-boundaries-algorithm-d8knq).

Thank you. I will try. Sorry to waste so much of your time

**Comment by Norserium** [comment]
@alexdpts, try [this example](https://codesandbox.io/s/vue-advanced-cropper-custom-stretcher-18mvp?file=/src/App.vue). It should provide the better user experience than `vue-js-clipper` for horizontal images.

**Comment by Norserium** [comment]
@alexdpts, any news?

**Comment by alexdpts** [comment]
> @alexdpts, any news?

Hello. Have some. 

The example with updateBoundaries works great within my markup (when cropper-wrapper is a flex cell) and set position:absolute;max-height:none; to cropper container . 

When trying to add (or replace with) the last example, need to wrap the cropper with a block with 
position: absolute; left: 0; top: 0; (otherwise, a third of the images goes beyond the upper border of the flex cell and is not available for cropping) everything seems to work (cropping, rotating, flipping), but when changing the image, errors appear in the console about the impossibility of determining naturalSize and aspectRatio of undefined (it looks like this is about starting stencil rendering);

Example with updateBoundaries is enough for me.

**Comment by Norserium** [comment]
@luckydonald what do you mean by "cropping takes some time"? Could you provide the video example?

Speaking of examples, could you provide the source of your example, that you posted above?

**Comment by alexdpts** [comment]
> @luckydonald, what do you mean by "cropping takes some time"? Could you provide the video example?

https://user-images.githubusercontent.com/17212759/127096611-bdd8eaae-9857-4609-89f4-b6fff316f054.mp4
(Clicking on a button occurs immediately on hover)
click handler source: 
```
const { canvas }=this.$root.$refs.cropper.getResult();
this.preview=canvas.toDataURL(1, 'image/jpeg');
```

> Speaking of examples, could you provide the source of your example, that you posted above?

Its your examples: 
[updateBoundaries](https://codesandbox.io/s/vue-advanced-cropper-custom-boundaries-algorithm-d8knq)  
and
[for horisontal images](https://codesandbox.io/s/vue-advanced-cropper-custom-stretcher-18mvp?file=/src/App.vue)

I'll make an example later

**Comment by Norserium** [comment]
> https://user-images.githubusercontent.com/17212759/127096611-bdd8eaae-9857-4609-89f4-b6fff316f054.mp4
(Clicking on a button occurs immediately on hover)
click handler source:

Thanks. I assume, that it should be implemented on your side. You can add the indicator right before calling the `getResult` method, and hide it in the end of the handler (it should be possible, because all the operations in your handler are synchronous).

But I don't recommend to implement the preview in this manner. 

First of all, the getting a result as `dataURL` produces large memory overhead and performance issues, it's better to use [`toBlob`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob) method.

Secondly, there is the much more fast solution: [`Preview`](https://norserium.github.io/vue-advanced-cropper/guides/recipes.html#preview-the-result)  component.

> later I will make an example with both of them in my layout

Yes, that's what I want to see. Because you've said before, that the problem reproduces only in your markup, not mine.

**Comment by Norserium** [comment]
@alexdpts, any news?

**Comment by alexdpts** [comment]
> @alexdpts, any news?

Not yet. No time. Later.

upd. Hmm. Start to make example, and all works fine (no errors in console). So somewhere I have a bug in my code. I will look for.
Thanks for your time and for great component.

**Comment by Norserium** [comment]
@alexdpts, any news?

**Comment by Norserium** [comment]
@alexdpts, any news?

**Comment by Norserium** [comment]
> How can i bind Zoom on mouseWheel+key (eq. Shift+mouseWheel)?

@alexdpts, there is the native possibility now. [The example](https://codesandbox.io/s/vue-advanced-cropper-custom-scroll-shift-xmwoh?file=/src/CustomBackgroundWrapper.vue).
